test/unit/local/steps/buffer.js

```diff
@@ -0,0 +1,372 @@
+/* eslint-env mocha */
+/* @flow */
+
+const Promise = require('bluebird')
+const _ = require('lodash')
+const should = require('should')
+const sinon = require('sinon')
+
+const Buffer = require('../../../../core/local/steps/buffer')
+
+const Builders = require('../../../support/builders')
+
+/*::
+import type {
+  AtomWatcherEvent,
+  Batch
+} from '../../../../core/local/steps/event'
+*/
+
+const builders = new Builders()
+
+function nonEmptyBatch (batchNumber /*: number */ = 1) /*: AtomWatcherEvent[] */ {
+  return [
+    builders.event().action('created').kind('file').path(`file-from-batch-${batchNumber}`).build(),
+    builders.event().action('deleted').kind('directory').path(`dir-from-batch-${batchNumber}`).build()
+  ]
+}
+
+should.Assertion.add('pending', function () {
+  this.params = {operator: 'be pending'}
+  this.obj.isPending().should.be.true()
+})
+
+describe('core/local/steps/Buffer', function () {
+  this.timeout(100)
+
+  describe('Basics', () => {
+    let buffer
+
+    beforeEach('instanciate', () => {
+      buffer = new Buffer()
+    })
+
+    describe('#push()', () => {
+      it('inserts the given Batch into the Buffer so it may be retrieved with #pop()', async () => {
+        const batch = nonEmptyBatch()
+
+        buffer.push(batch)
+
+        await should(buffer.pop()).be.fulfilledWith(batch)
+      })
+
+      it('inserts the same batch multiple times', async () => {
```

Comment:
The way I understand this description is a call to `push()` will insert the given batch multiple times.
What do you think of "does not squash equivalent batches" ?